contracts/gateway/src/contract.rs

```diff
@@ -61,40 +50,37 @@ pub fn execute(
 
 pub mod execute {
 
-    use cosmwasm_std::{to_binary, QueryRequest, StdError, WasmMsg, WasmQuery};
+    use connection_router::state::CrossChainId;
+    use cosmwasm_std::{to_binary, QueryRequest, WasmMsg, WasmQuery};
 
     use crate::{events::GatewayEvent, state::OUTGOING_MESSAGES};
 
     use super::*;
 
-    fn contains_duplicates(msgs: &mut Vec<Message>) -> bool {
+    fn contains_duplicates(msgs: &mut Vec<NewMessage>) -> bool {
         let orig_len = msgs.len();
-        msgs.sort_unstable_by_key(|a| a.id.to_string());
-        msgs.dedup_by(|a, b| a.id == b.id);
+        msgs.sort_unstable_by_key(|a| a.cc_id.to_string());
+        msgs.dedup_by(|a, b| a.cc_id == b.cc_id);
         orig_len != msgs.len()
```

Comment:
```suggestion
        msgs.iter()
            .map(|msg| msg.cc_id.clone())
            .collect::<HashSet<_>>()
            .len()
            != msgs.len()
```

Reply:
The sorting seems unnecessarily costly.

Reply:
I find you're way cleaner @sammy1991106 , but sorting actually might be faster here, because it's probably a small vector, and the sorting is in place

Reply:
I left this as is on purpose, because dealing with duplicates is a separate issue on my board, and for that this whole message changes 😆

Reply:
fair enough. I can't see how sorting is more efficient though. It's O(NlogN) v.s. O(N) to me.